llama_hub/llama_packs/multi_tenancy_rag/base.py

```diff
@@ -21,13 +25,24 @@ def get_modules(self) -> Dict[str, Any]:
 
     def add(self, documents: List[Document], user: Any) -> None:
         """Insert Documents of a user into index"""
+        # Add metadata to documents
         for document in documents:
             document.metadata["user"] = user
-            self.index.insert(document)
+        # Create Nodes using IngestionPipeline
+        pipeline = IngestionPipeline(
+            transformations=[
+                SentenceSplitter(chunk_size=512, chunk_overlap=20),
+            ]
+        )
+        nodes = pipeline.run(documents=documents)
```

Comment:
if have a long list of documents might want to consider using parallel processing, by setting `num_workers` to a positive integer value, like 4.

Reply:
Sure. Updated